source/adios2/helper/adiosKokkos.cpp

```diff
@@ -98,6 +92,15 @@ bool IsGPUbuffer(const void *ptr)
     {
         return true;
     }
+#endif
+#ifdef ADIOS2_HAVE_KOKKOS_SYCL
+    auto ret =
+        sycl::address_space_cast<sycl::access::address_space::local_space,
+                                 sycl::access::decorated::no>(ptr);
```

Comment:
That should probably be
```suggestion
        sycl::address_space_cast<sycl::access::address_space::global_space,
                                 sycl::access::decorated::no>(ptr);
```
assuming that you expect the allocations to come from sycl::malloc. You would use `local_space` for scratch memory. If you care about testing for host accessibility you would refine to `ext_intel_global_device_space` resp. `ext_intel_global_host_space`, see https://github.com/intel/llvm/blob/f363bb272b5a85849059fff40c3b321910f87781/sycl/doc/extensions/supported/sycl_ext_intel_usm_address_spaces.asciidoc#sycl-device-memory-model.

Reply:
We are using `std::copy` to cache the data in an internal ADIOS buffer outside this function. I want to use `address_space_cast` to know when there is no way to use `std::copy` in which case I fall back to Kokkos. I was assuming `global_space` would still allow the Host to access it.

Reply:
Makes sense, thanks. Changed.